pkg/controllers/streaming/processor_controller.go

```diff
@@ -667,12 +669,16 @@ func (*ProcessorReconciler) collectAliases(bindings []streamingv1alpha1.StreamBi
 }
 
 func (r *ProcessorReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	enqueueTrackedResources := func(t apis.Resource) handler.EventHandler {
+	enqueueTrackedResources := func(t runtime.Object) handler.EventHandler {
+		versionKinds, _, err := r.Scheme.ObjectKinds(t)
+		if err != nil {
+			panic(err)
+		}
 		return &handler.EnqueueRequestsFromMapFunc{
 			ToRequests: handler.ToRequestsFunc(func(a handler.MapObject) []reconcile.Request {
-				requests := []reconcile.Request{}
+				var requests []reconcile.Request
 				key := tracker.NewKey(
-					t.GetGroupVersionKind(),
+					versionKinds[0],
```

Comment:
Maybe we should also panic above if len(versionKinds) != 1

Reply:
> Maybe we should also panic above if len(versionKinds) != 1

There can be multiple valid GroupVersions for a single type. Ingress is in `networking.k8s.io/v1beta1 ` and `extensions/v1beta1`.